tests/shared/drive-constraints.spec.js

```diff
@@ -556,4 +556,266 @@ describe('Shared: DriveConstraints', function() {
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.OK,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.OK
+        });
+      });
+
+    });
+
+    describe('given the drive is null', () => {
+
+      it('should return OK', function() {
+        const result = constraints.getDriveImageCompatibilityStatus(null, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000,
+          recommendedDriveSize: 2000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.OK,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.OK
+        });
+      });
+
+    });
+
+    describe('given a system drive and image is null', () => {
+
+      it('should return system drive error', function() {
+        this.drive.system = true;
+
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, null);
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.WARNING,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.SYSTEM
+        });
+      });
+
+    });
+
+    describe('given a locked drive and image is null', () => {
+
+      it('should return locked drive error', function() {
+        this.drive.protected = true;
+
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, null);
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.ERROR,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.LOCKED
+        });
+      });
+
+    });
+
+    describe('given the drive contains the image and is a system drive', () => {
+
+      it('should return the source drive error by precedence', function() {
+        this.drive.system = true;
+
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: `${this.mountpoint}${this.separator}rpi.img`,
+          size: 1000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.ERROR,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.CONTAINS_IMAGE
+        });
+      });
+
+    });
+
+    describe('given a system and too small drive', () => {
+
+      it('should return the system drive error by precedence', function() {
+        this.drive.system = true;
+
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 5000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.WARNING,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.SYSTEM
+        });
+      });
+
+    });
+
+    describe('given a too small and locked drive', () => {
+
+      it('should return the too small error by precedence', function() {
```

Comment:
Maybe a "locked" error should be considered more important than a "too small" error?